### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by st60994**
Václav Hrubý

### CODE REVIEW

File: NNPTPZ1/ComplexNumber.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Mathematics
+{
+    public class ComplexNumber
+    {
+        public double RealPart { get; set; }
+        public float ImaginaryPart { get; set; }
```
Comment (rodi0878) [inline NNPTPZ1/ComplexNumber.cs]: real as double and imaginary as float is nonsense

File: NNPTPZ1/ComplexNumber.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        {
+            return Math.Sqrt(RealPart * RealPart + ImaginaryPart * ImaginaryPart);
+        }
+
+        /// <summary>
+        /// Sums up current ComplexNumber with the one provided in the parameter and creates a new one
+        /// </summary>
+        /// <param name="b"></param>
+        /// <returns></returns>
+        public ComplexNumber Sum(ComplexNumber b)
```
Comment (rodi0878) [inline NNPTPZ1/ComplexNumber.cs]: Add not Sum

File: NNPTPZ1/ComplexNumber.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        {
+            ComplexNumber a = this;
+            return new ComplexNumber()
+            {
+                RealPart = a.RealPart + b.RealPart,
+                ImaginaryPart = a.ImaginaryPart + b.ImaginaryPart
+            };
+        }
+        public double GetAngleInDegrees()
```
Comment (rodi0878) [inline NNPTPZ1/ComplexNumber.cs]: Really Math.Atan returns degrees?

File: NNPTPZ1/FractalVisualiser.cs [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        private void colorizePixel(int x, int y, int numberOfTotalIterations, int actualRootPosition)
+        {
+            var vv = colors[actualRootPosition % colors.Length];
+            vv = Color.FromArgb(Math.Min(Math.Max(0, vv.R - numberOfTotalIterations * 2), 255), Math.Min(Math.Max(0, vv.G - numberOfTotalIterations * 2), 255), Math.Min(Math.Max(0, vv.B - numberOfTotalIterations * 2), 255));
+            bitmap.SetPixel(x, y, vv);
+        }
+
+        private static int findPositionOfActualRootInArray(List<ComplexNumber> roots, ComplexNumber functionRoot)
+        {
+            var actualRootFound = false;
```
Comment (rodi0878) [inline NNPTPZ1/FractalVisualiser.cs]: var is not a good option here

File: NNPTPZ1/FractalVisualiser.cs [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+            if (functionRoot.RealPart == 0)
+                functionRoot.RealPart = 0.0001;
+            if (functionRoot.ImaginaryPart == 0)
+                functionRoot.ImaginaryPart = 0.0001f;
+            return functionRoot;
+        }
+
+        private void colorizePixel(int x, int y, int numberOfTotalIterations, int actualRootPosition)
+        {
+            var vv = colors[actualRootPosition % colors.Length];
```
Comment (rodi0878) [inline NNPTPZ1/FractalVisualiser.cs]: vv?

File: NNPTPZ1/Polynom.cs [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+                evaluatedPolynom = evaluatedPolynom.Sum(coefficient);
+            }
+
+            return evaluatedPolynom;
+        }
+        public override string ToString()
+        {
+            string s = "";
+            int i = 0;
```
Comment (rodi0878) [inline NNPTPZ1/Polynom.cs]: thats ugly.

**Review by rodi0878 — CHANGES_REQUESTED** [review]
Please look at attached comments and provide update

**Comment by st60994** [comment]
Update with required changes has been provided.

**Comment by rodi0878** [comment]
Accepted

**Pull request closed without merging**